usage_analytics/analytics_http_client.py

```diff
@@ -0,0 +1,39 @@
+import httpx
+from .analytics_definition import ANALYTICS_TO_TRACK
+
+
+class AnalyticsHttpClient(httpx.Client):
+    """
+    Custom HTTP client that automatically adds the calling class to all requests.
```

Comment:
Why is a client in `common`?  
Shouldn't this be in client only?

Reply:
Yeah good catch there.
My initial implementation involves exposing AnalyticsHttpClient, but I changed it afterwards. I'll move this accordingly.

Reply:
I thought about it again and tried different variants, I think it may be a better idea to keep the `AnalyticsHttpClient` here. Here's my train of reasoning
  - `AnalyticsHttpClient` has a specific behavior when the header request is `PL-Module-Name`
  - `PL-Module-Name` is defined in the `ANALYTICS_HEADER_CONFIG`
  - `AnalyticsHttpClient` and `ANALYTICS_HEADER_CONFIG` should be in the same repo
  - `ANALYTICS_HEADER_CONFIG` has to be in `tabpfn_common_utils`
  - Hence `AnalyticsHttpClient` should be in `tabpfn_common_utils`